workflow/controller/operator.go

```diff
@@ -412,8 +412,8 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 	}
 
 	var workflowMessage string
-	if node.FailedOrError() && woc.execWf.Spec.Shutdown != "" {
```

Comment:
can I ask you to add `woc.GetShutdownStrategy()` - that way we can refactor later

Reply:
Added `needsStoredWfSpecUpdate()` determine to update the `storedWorkflowSpec`.